Source/Scene/PointCloud3DTileContent.js

```diff
@@ -788,58 +693,294 @@ define([
+            styleTranslucent = info.translucent;
+        }
+
+        content._styleTranslucent = styleTranslucent;
+
+        var hasColorStyle = defined(colorStyleFunction);
+        var hasShowStyle = defined(showStyleFunction);
+
+        // Get the properties in use by the style
+        var styleableProperties = [];
+        if (hasColorStyle) {
+            getStyleableProperties(colorStyleFunction, styleableProperties);
+        }
+        if (hasShowStyle) {
+            getStyleableProperties(showStyleFunction, styleableProperties);
+        }
+
+        // Disable vertex attributes that aren't used in the style, enable attributes that are
+        var styleableShaderAttributes = content._styleableShaderAttributes;
+        if (defined(styleableShaderAttributes)) {
+            for (name in styleableShaderAttributes) {
+                if (styleableShaderAttributes.hasOwnProperty(name)) {
+                    attribute = styleableShaderAttributes[name];
+                    var enabled = (styleableProperties.indexOf(name) >= 0);
+                    var vertexAttribute = getVertexAttribute(vertexArray, attribute.location);
+                    vertexAttribute.enabled = enabled;
+                }
+            }
+        }
+
+        if (hasColorStyle) {
+            // Disable the color vertex attribute when a color style is used
+            var colorVertexAttribute = getVertexAttribute(vertexArray, colorLocation);
+            colorVertexAttribute.enabled = false;
+        }
+
+        var attributeLocations = {
+            a_position : positionLocation
+        };
+        if (hasColors && !hasColorStyle) {
+            attributeLocations.a_color = colorLocation;
+        }
+        if (hasNormals) {
+            attributeLocations.a_normal = normalLocation;
+        }
+        if (hasBatchIds) {
+            attributeLocations.a_batchId = batchIdLocation;
+        }
+
+        var attributeIncludes = '';
+        var length = styleableProperties.length;
+        for (i = 0; i < length; ++i) {
+            name = styleableProperties[i];
+            attribute = styleableShaderAttributes[name];
+
+            //>>includeStart('debug', pragmas.debug);
+            if (!defined(attribute)) {
+                throw new DeveloperError('Style reference a property "' + name + '" that does not exist or is not styleable.');
+            }
+            //>>includeEnd('debug');
+
+            var componentCount = attribute.componentCount;
+            var attributeName = 'czm_tiles3d_style_' + name;
+            var attributeType;
+            if (componentCount === 1) {
+                attributeType = 'float';
+            } else {
+                attributeType = 'vec' + componentCount;
+            }
+
+            attributeIncludes += 'attribute ' + attributeType + ' ' + attributeName + '; \n';
+            attributeLocations[attributeName] = attribute.location;
+        }
+
+        var vs = 'attribute vec3 a_position; \n' +
```

Comment:
Addressed in https://github.com/AnalyticalGraphicsInc/cesium/pull/4336/commits/b7c2dfddb58a1477e6023f32e314668541106957, though I'm not sure if you had something else in mind.